cmd/kube-proxy/app/server_others.go

```diff
@@ -50,6 +50,28 @@ import (
 	"k8s.io/klog"
 )
 
+// Reduces a list of bindaddresses down to the first ipv4 and first ipv6 address in the list.
+// (Doesn't check that IP addresses are valid)
+// Returns: []{<ipv4 || ipv6>} || []{<ipv4, ipv6>} || []{<ipv6, ipv4>}
+func parseBindAddresses(addresses []string) []string {
+	filteredAddresses := make([]string, 0)
+
+	hasV4 := false
+	hasV6 := false
+
+	for _, addr := range addresses {
+		if !hasV4 && net.ParseIP(addr).To4() != nil {
```

Comment:
You probably will need to do that in a lot of places. Let us use https://github.com/kubernetes/utils/blob/master/net/net.go#L113 it is already vendored in k/k